vision/api/VisionTest/VisionTest.cs

```diff
@@ -195,6 +197,25 @@ public void DetectCropHint()
             Assert.Contains("\tX:", output.Stdout);
             Assert.Contains("\tY:", output.Stdout);
         }
+
+        [Fact]

```

Comment:
Move this test into its own class

Every test in `CommonTests` is run 3 times with various arguments provided (once to test using local .png file, once to test using http:// url to .png, once to test using gs:// url to .png)

We don't want that :)

You could add it right below, class... `PdfOcrTests`... with one `DetectDocument()` method, as is... then you can could a line to have it upload the .pdf file before the test runs or something :)

Reply:
Example: remove the current `DetectDocument` method from `CommonTests` and add the following class 
after `TutorialTests` before `CommonTests`

```cs
// TutorialTests
// ...

/// <summary>
/// Runs tests detecting text from PDF or TIFF files
/// These require uploading a .pdf to Google Cloud Storage
/// and then API results are saved to Google Cloud Storage
/// </summary>
public class PdfDocumentTests : IClassFixture<RandomBucketFixture>
{
  readonly string _bucketName;
  readonly BucketCollector _bucketCollector;
  readonly string _pdfFileName = "HodgeConj.pdf";

  public PdfDocumentTests(RandomBucketFixture bucketFixture)
  {
    _bucketName = bucketFixture.BucketName;
    _bucketCollector = new BucketCollector(_bucketName);
    var localPath = Path.Combine("data", _pdfFileName)
    _bucketCollector.CopyToBucket(localPath, _pdfFileName);
  }

  [Fact]
  public void TestDetectPdfDocument()
  {
    var outputPrefix = "OCR_PDF_TEST_OUTPUT";
    var gcsSourceURI = $"gs://{_bucketName}/{_pdfFileName}";
    var gcsDestinationURI = $"gs://{_bucketName}/{outputPrefix}/";

    var output = Run("ocr", gcsSourceURI, gcsDestinationURI);
    var storageClient = StorageClient.Create();
    var bucket = storageClient.GetBucket(_bucketName);
    var blobList = storageClient.ListObjects(bucketName, outputPrefix);

    Assert.Equal(0, output.ExitCode);
    Assert.Contains("Full text:", output.Stdout);
    Assert.Contains("Hodge conjecture", output.Stdout);
    Assert.Equal(blobList.Count(), 1);
  }
  
  public void Dispose() => _bucketCollector.Dispose();
}

// ...
// CommonTests
```

Reply:
... if this blows up, you might have to do the following, I didn't have intellisense, just typed this out ...

```cs
public void Dispose() => ((IDisposable)_bucketCollector).Dispose();
```

Reply:
FYI References
 - [GCS Test Helper file](https://github.com/GoogleCloudPlatform/dotnet-docs-samples/blob/b573a1402310f6ebc089cbdd7666211f275dbf97/testutil/TestBucket.cs) – it's only ~ 100 LOC with the following 2 helper classes
   - [RandomBucketFixture](https://github.com/GoogleCloudPlatform/dotnet-docs-samples/blob/b573a1402310f6ebc089cbdd7666211f275dbf97/testutil/TestBucket.cs#L26)
   - [BucketCollector](https://github.com/GoogleCloudPlatform/dotnet-docs-samples/blob/b573a1402310f6ebc089cbdd7666211f275dbf97/testutil/TestBucket.cs#L62)